examples/nodejs/lambda-examples/cloudwatch-metrics/infrastructure/lib/momento-metrics-stack.ts

```diff
@@ -0,0 +1,337 @@
+      dashboardName: 'MomentoMetricsCDKExampleDashboard',
+      defaultInterval: cdk.Duration.hours(1),
+    });
+    this.addWidgetsToDashboard(dashboard);
+  }
+
+  setUpLambdaFunction(apiKeySecret: cdk.aws_secretsmanager.Secret) {
+    const nodejsLambda = new lambdaNodejs.NodejsFunction(this, 'MomentoMetricsMiddlewareCDKExample', {
+      functionName: 'MomentoMetricsMiddlewareCDKExample',
+      runtime: lambda.Runtime.NODEJS_LATEST,
+      entry: path.join(__dirname, '../../lambda/handler.ts'),
+      projectRoot: path.join(__dirname, '../../lambda'),
+      depsLockFilePath: path.join(__dirname, '../../lambda/package-lock.json'),
+      handler: 'handler',
+      timeout: cdk.Duration.minutes(6),
+      memorySize: 128,
+      environment: {
+        MOMENTO_API_KEY_SECRET_NAME: apiKeySecret.secretName,
+      },
+    });
+    apiKeySecret.grantRead(nodejsLambda);
+  }
+
+  setUpEcsCluster(logGroup: cdk.aws_logs.LogGroup, apiKeySecret: cdk.aws_secretsmanager.Secret) {
+    const cluster = new ecs.Cluster(this, 'MomentoMetricsExampleFargateCluster');
+
+    const imageAsset = new DockerImageAsset(this, "MomentoMetricsECSDockerImage", {
+      directory: path.join(__dirname, "../../docker")
+    });
+
+    const taskDefinition = new ecs.FargateTaskDefinition(this, 'MomentoMetricsMiddlewareCDKExample');
+    taskDefinition.addContainer('MomentoMetricsECSContainer', {
+      image: ecs.ContainerImage.fromDockerImageAsset(imageAsset),
+      logging: new ecs.AwsLogDriver({
+        logGroup: logGroup,
+        streamPrefix: "MomentoMetricsMiddlewareCDKExample"
+      })
+    });
+
+    const policy = new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      actions: [
+        "secretsmanager:GetSecretValue",
+        "logs:PutLogEvents",
+        "logs:CreateLogStream",
+        "ecs:*"
+      ],
+    });
+    policy.addResources(
+      apiKeySecret.secretArn, 
+      logGroup.logGroupArn, 
+      cluster.clusterArn, 
+      taskDefinition.taskDefinitionArn
+    );
+    taskDefinition.addToTaskRolePolicy(policy);
+    taskDefinition.addToExecutionRolePolicy(policy);
+
+    new ecs.FargateService(this, 'MomentoMetricsECSFargateService', {
+      cluster,
+      taskDefinition,
+    });
+  }
+
+  createMetricFilters(logGroup: cdk.aws_logs.LogGroup) {
+    logGroup.addMetricFilter('ExampleMetricFilterDuration', {
+      metricNamespace: 'MomentoMetricsCDKExample',
+      metricName: 'Duration (Latency)',
+      filterPattern: FilterPattern.literal('{$.duration > 0}'),
+      metricValue: '$.duration',
+      unit: Unit.MILLISECONDS,
+    });
+
+    logGroup.addMetricFilter('ExampleMetricFilterRequestSize', {
+      metricNamespace: 'MomentoMetricsCDKExample',
+      metricName: 'Request Size (bytes)',
+      filterPattern: FilterPattern.literal('{$.requestSize >= 0}'),
+      metricValue: '$.requestSize',
+      unit: Unit.BYTES,
+    });
+
+    logGroup.addMetricFilter('ExampleMetricFilterResponseSize', {
```

Comment:
see my comments below about `dimensions` before reading this.

For this metric, adding a dimension for the request type would be extremely useful. Then we could graph p50/p99 request size / response size by request type, for a few of the most common request types.

This also would rely on fixing the bug I mentioned above about how the request type is currently showing up as 'middleware' or whatever.